src/components/GoodsList.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+type State = {
+  goods: string[],
+};
+
+type Props = {
+  goods: string[],
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    goods: this.props.goods,
+  };
+
+  reverseGoods = () => {
+    return (
+      this.setState(state => (
+        {
+          goods: [...state.goods].reverse(),
+        }
+      ))
+    );
+  };
+
+  sortGoodsAlphabetically = () => {
+    return (
+      this.setState(state => (
+        {
+          goods: [...state.goods].sort(),
```

Comment:
Better to use locale.Compare()

Reply:
agree, use localeCompare for sorting strings